src/ripple/app/paths/PathRequests.cpp

```diff
@@ -97,17 +120,26 @@ PathRequests::updateAll(
                     remove = false;
                 else
                 {
-                    if (auto ipSub = request->getSubscriber())
+                    if (auto ipSub = getSubscriber(request))
                     {
                         if (!ipSub->getConsumer().warn())
                         {
-                            Json::Value update =
-                                request->doUpdate(cache, false);
+                            // Release the shared ptr to the subscriber so that
+                            // it can be freed if the client disconnects, and
+                            // thus fail to lock later.
+                            ipSub.reset();
+                            Json::Value update = request->doUpdate(
+                                cache, false, [&getSubscriber, &request]() {
+                                    return (bool)getSubscriber(request);
+                                });
```

Comment:
It's not obvious what that lambda is doing. Maybe make a local variable for the callback just for the purpose of nameing it? I.e.
```
auto conintueCallback = [&getSubscriber, &request]()...
```

Reply:
Done.